envoy/router/router.h

```diff
@@ -530,12 +530,55 @@ using ShadowPolicyPtr = std::shared_ptr<ShadowPolicy>;
   STATNAME(vcluster)                                                                               \
   STATNAME(vhost)
 
+/**
+ * All route level stats. @see stats_macro.h
+ */
+#define ALL_ROUTE_STATS(COUNTER, GAUGE, HISTOGRAM, TEXT_READOUT, STATNAME)                         \
+  COUNTER(upstream_rq_retry)                                                                       \
+  COUNTER(upstream_rq_retry_limit_exceeded)                                                        \
+  COUNTER(upstream_rq_retry_overflow)                                                              \
+  COUNTER(upstream_rq_retry_success)                                                               \
+  COUNTER(upstream_rq_timeout)                                                                     \
+  COUNTER(upstream_rq_total)                                                                       \
+  STATNAME(route)                                                                                  \
+  STATNAME(vhost)
+
 /**
  * Struct definition for all virtual cluster stats. @see stats_macro.h
  */
 MAKE_STAT_NAMES_STRUCT(VirtualClusterStatNames, ALL_VIRTUAL_CLUSTER_STATS);
 MAKE_STATS_STRUCT(VirtualClusterStats, VirtualClusterStatNames, ALL_VIRTUAL_CLUSTER_STATS);
 
+/**
+ * Struct definition for all route level stats. @see stats_macro.h
+ */
+MAKE_STAT_NAMES_STRUCT(RouteStatNames, ALL_ROUTE_STATS);
+MAKE_STATS_STRUCT(RouteStats, RouteStatNames, ALL_ROUTE_STATS);
+
+/**
+ * RouteStatsContext defines config needed to generate all route level stats.
+ */
+struct RouteStatsContext {
+private:
+  const Stats::StatNameManagedStorage route_stat_name_storage_;
```

Comment:
make this a class and not a struct, and put the public block before the private block, per convention.

https://google.github.io/styleguide/cppguide.html#Declaration_Order